src/odemis/acq/stream/_base.py

```diff
@@ -848,43 +848,40 @@ def _shouldUpdateImage(self):
         self._im_needs_recompute.set()
 
     @staticmethod
-    def _image_thread(wstream):
+    def _image_thread(wstream, period=0.1):
         """ Called as a separate thread, and recomputes the image whenever it receives an event
         asking for it.
 
         Args:
             wstream (Weakref to a Stream): the stream to follow
-
+            period ( float > 0) (Seconds): Minimum time in second between two image updates
         """
-
         try:
             stream = wstream()
             name = stream.name.value
             im_needs_recompute = stream._im_needs_recompute
             # Only hold a weakref to allow the stream to be garbage collected
             # On GC, trigger im_needs_recompute so that the thread can end too
             wstream = weakref.ref(stream, lambda o: im_needs_recompute.set())
-
-            tnext = 0
             while True:
                 del stream
+                #TODO K.K. Eric does this go alright with waiting/pointers etc?
```

Comment:
That looks fine. Especially, assuming it was fine, the changes shouldn't bring more issues.